front/lib/resources/storage/models/data_source.ts

```diff
@@ -15,7 +15,7 @@ export class DataSourceModel extends SoftDeletableModel<DataSourceModel> {
   declare updatedAt: CreationOptional<Date>;
 
   // Corresponds to the ID of the last user to configure the connection.
-  declare editedByUserId: ForeignKey<User["id"]>;
+  declare editedByUserId: ForeignKey<User["id"]> | null;
```

Comment:
@flvndvd typing questions here. The types in `front/lib/resources/types.ts` are unhappy with adding a `?` in front of editedByUser below:
```
declare editedByUser?: NonAtrtibute<User>
```
Which is a pattern we've used elesewhere (eg Conversations).

All that being said even when `includeEditedBy` is set to `true` everything works as expected (tested poke data source / data source views list and knowledge folders list which both request it).

The typing already contemplated having editedBy to undefined so product pathes are already handled.

Just want to make sure you're happy with the diff to `front/lib/resources/types.ts` (below) and the way it's used in both resources in this PR.

Reply:
In SQL, there are optional types, it's materialized through `nullable`. If we attempt to use `?`, it would be an error because database types are either null or a specific type like number. Here we want the TypeScript type to align with these database types.

Reply:
@flvndvd was referring not to this type but the directive below `declare editedByUser: NonAttribute<User>;`

Reply:
Will move forward with this but want to make sure we're aligned :+1: